site/src/hooks/usePrefersReduceMotion.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react'
+
+const query = '(prefers-reduced-motion: no-preference)'
+
+function getPrefersReducedMotion() {
+  if (typeof window === 'undefined') {
+    return false
+  }
+
+  const mediaQueryList = window.matchMedia(query)
+
+  return !mediaQueryList.matches
+}
+
+export function usePrefersReducedMotion() {
+  const [prefersReducedMotion, setPrefersReducedMotion] = useState(getPrefersReducedMotion)
+
+  useEffect(() => {
+    const mediaQueryList = window.matchMedia('(prefers-reduced-motion: no-preference)')
+    const listener = () => {
+      setPrefersReducedMotion(getPrefersReducedMotion)
+    }
+
+    mediaQueryList.addEventListener('change', listener)
+    return () => {
+      mediaQueryList.addEventListener('change', listener)
```

Comment:
i think here should be something like removeEventListener

Reply:
OMG, crazy bug

Reply:
Done